content/200-concepts/100-components/02-prisma-client/064-custom-models.mdx

```diff
@@ -0,0 +1,93 @@
+---
+title: 'Custom Models'
+metaTitle: 'Custom Models'
+metaDescription: 'This page explains how to wrap the Prisma Client in custom models'
+---
+
+<TopBlock>
+
+As your application grows, you may find the need to group related logic together.
+
+We suggest either wrapping a model in a class or extending the Prisma Client model object.
+
+</TopBlock>
+
+## Wrap a Prisma Model in a Class
+
+In the example below, we're wrapping the `user` model in the Prisma Client within a `Users` class. This class adds a custom signup function.
+
+```tsx highlight=4,23,28-30;normal
+import { PrismaClient } from '@prisma/client'
+
+type Signup = {
+  email: string
+  firstName: string
+  lastName: string
+}
+
+class Users {
+  constructor(private readonly prisma: PrismaClient['user']) {}
+
+  // Signup a new user
+  async signup(data: Signup): Promise<User> {
+    // do some custom validation...
+    return this.prisma.create({ data })
+  }
+}
+
+async function main() {
+  const prisma = new PrismaClient()
+  const users = new Users(prisma.user)
+  const user = await users.signup({
+    email: 'alice@prisma.io',
+    firstName: 'Alice',
+    lastName: 'Prisma',
+  })
+}
+```
+
+Here we're only exposing a `signup` method from the Prisma Client. With this approach, you'll no longer be able to call Prisma Client methods like `findMany` or `upsert`. This approach works well when you have a large application and you want to intentionally limit what your models can do.
+
```

Comment:
By 'Here" do you mean the section above, right? 
Should we briefly explain why wrapping `user` model in the `User` class disallows you from call the methods mentioned? Does wrapping mean that only what the class allows is available, and the only thing that class allows/provides is the signup feature? Or maybe this is super-obvious to even newbie developers?

Reply:
Just tried to clarify a bit